src/util/TchapClientUtils.ts

```diff
@@ -0,0 +1,99 @@
+import BasePlatform from "matrix-react-sdk/src/BasePlatform";
+import { MatrixClientPeg } from "matrix-react-sdk/src/MatrixClientPeg";
+import PlatformPeg from "matrix-react-sdk/src/PlatformPeg";
+
+export default class TchapClientUtils {
+    static SYNC_STORE_NAME = "matrix-js-sdk:riot-web-sync";
+
+    /**
+     * Determine weither the app needs a refresh after loading
+     * @param indexedDB the indexDb interface
+     * @returns Promise(true) if a refresh is needed, Promise(false) in other cases
+     */
+    public static async doesNeedRefresh(indexedDB: IDBFactory): Promise<boolean> {
```

Comment:
This function has "4" hardcoded, so I suggest :
```suggestion
    public static async needsRefreshForVersion4(indexedDB: IDBFactory): Promise<boolean> {
```
(another option is to make a customiseable function that we can reuse in the future if this happens again... but I think keeping it simple for now is a good idea to avoid problems.)